substrate/client/db/src/pinned_blocks_cache.rs

```diff
@@ -20,7 +20,7 @@ use schnellru::{Limiter, LruMap};
 use sp_runtime::{traits::Block as BlockT, Justifications};
 
 const LOG_TARGET: &str = "db::pin";
-const PINNING_CACHE_SIZE: usize = 1024;
+const PINNING_CACHE_SIZE: usize = 2048;
```

Comment:
This should be larger than the limit set in the notification pinning worker, since this is the maximum number of pinned block bodies overall: Notifications + RPC API + other uses. The number of pins in the state-db is not limited at all, it trusts users entirely to unpin by themselves. Here it is left it as a last resort.
If you don't like the value specifically, we can reduce both the values here and in the worker.